source/particle/world.cc

```diff
@@ -553,28 +558,64 @@ namespace aspect
                           ExcMessage("Periodic boundaries combined with particles currently "
                                      "only work with 90 degree opening angle in spherical shell."));
 
+              // Find out if we need to rotate the temporary properties
+              unsigned int property_index_to_rotate = numbers::invalid_unsigned_int;
+              if (Plugins::plugin_type_matches<Particle::Integrator::Euler<dim>>(*integrator) == true)
+                {
+                  // nothing to do
+                }
+              else if (Plugins::plugin_type_matches<Particle::Integrator::RK2<dim>>(*integrator) == true)
+                {
+                  // rotate original location
+                  property_index_to_rotate = get_property_manager().get_data_info().get_position_by_field_name("internal: integrator properties");
+                  AssertThrow(property_index_to_rotate != numbers::invalid_unsigned_int, ExcMessage("Could not find the property index."));
+                }
+              else
+                {
+                  AssertThrow(false,
+                              ExcMessage("Periodic boundaries in a spherical shell are not supported for this integration scheme."));
+                }
+
               typename ParticleHandler<dim>::particle_iterator particle = particle_handler->begin();
               for (; particle != particle_handler->end(); ++particle)
                 {
                   // modify the particle position if it crossed a periodic boundary
                   Point<dim> particle_position = particle->get_location();
 
+                  // define a rotation matrix for the new position depending on the boundary
+                  Tensor<2,dim> rotation_matrix;
```

Comment:
Can you explain this a bit? If I understand your code correctly, what you're really trying to do is move a particle that has moved out of one side of the domain back in from the other side, and because (i) the domain is a quarter shell and (ii) the origin is the center of the shell, this corresponds to rotating the point around the origin by a certain amount? I think this kind of explanation would be useful.

At the same time, I don't understand why this is correct. I can see that this works if a particle moves out of one side and back in on the other side, but not the other way around. What do you do there?